### PR TITLE
chore: Adding test Upgrade for 4 tests

### PR DESCRIPTION
**Pull request by akshaybathija-github**
Adding New Framework tests for 
 - Shared VPC
 - Simple Autopilot Private
 - Simple Autopilot Public 
 - Simple Regional

### CODE REVIEW

File: test/integration/simple_autopilot_private/simple_autopilot_private_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		serviceAccount := bpt.GetStringOutput("service_account")
+
+		op := gcloud.Runf(t, "container clusters describe %s --zone %s --project %s", clusterName, location, projectId)
+		g := golden.NewOrUpdate(t, op.String(),
+			golden.WithSanitizer(golden.StringSanitizer(serviceAccount, "SERVICE_ACCOUNT")),
+			golden.WithSanitizer(golden.StringSanitizer(projectId, "PROJECT_ID")),
+			golden.WithSanitizer(golden.StringSanitizer(clusterName, "CLUSTER_NAME")),
+		)
+		validateJSONPaths := []string{
+			"status",
```
Comment (bharathkkb) [inline test/integration/simple_autopilot_private/simple_autopilot_private_test.go:44]: This sometimes is in reconciling state which causes flakes (https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1301). Could we do a custom assert for just this field to check either RUNNING or RECONCILING?

File: test/integration/simple_autopilot_public/simple_autopiliot_public_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		serviceAccount := bpt.GetStringOutput("service_account")
+
+		op := gcloud.Runf(t, "container clusters describe %s --zone %s --project %s", clusterName, location, projectId)
+		g := golden.NewOrUpdate(t, op.String(),
+			golden.WithSanitizer(golden.StringSanitizer(serviceAccount, "SERVICE_ACCOUNT")),
+			golden.WithSanitizer(golden.StringSanitizer(projectId, "PROJECT_ID")),
+			golden.WithSanitizer(golden.StringSanitizer(clusterName, "CLUSTER_NAME")),
+		)
+		validateJSONPaths := []string{
+			"status",
```
Comment (bharathkkb) [inline test/integration/simple_autopilot_public/simple_autopiliot_public_test.go:44]: Same here, let's do a custom assert for just this field to check either RUNNING or RECONCILING

**Review by bharathkkb — COMMENTED** [review]
Thanks @akshaybathija-github 
A minor nit

File: test/integration/simple_autopilot_private/simple_autopilot_private_test.go (hunk 1 above)
Comment (akshaybathija-github) [reply test/integration/simple_autopilot_private/simple_autopilot_private_test.go:44]: in either scenario it would work but fail in case of a third state

Comment (akshaybathija-github) [reply test/integration/simple_autopilot_private/simple_autopilot_private_test.go]: Modified Assert Condition

File: test/integration/simple_autopilot_public/simple_autopiliot_public_test.go (hunk 2 above)
Comment (akshaybathija-github) [reply test/integration/simple_autopilot_public/simple_autopiliot_public_test.go]: Modified Assert Condition

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**